### ISSUE TITLE
Youtube Laggy with Hardware Acceleration if Twitch Streams Open any browser.

### ISSUE THREAD

**Issue by thesunnymachine** [Alex313031/Thorium-Win#103]
**System Details**
 - OS: Windows 10
 - Version 117.0.5938.157 (Official Build) (64-bit)

**Problem**
If I have ANY Browser open running twitch streams and I am watching youtube on Thorium, the Youtube video will get out sync, buffer, and struggle to play correctly. 

Turning off Hardware acceleration fixes the issue.  But this issue does not happen with Edge or Chrome at all. 

**Screenshots**
NA

**Additional Notes**

**Comment by gz83** [comment]
Please check whether the graphics card driver is normal and is the latest version

**Comment by thesunnymachine** [comment]
> Please check whether the graphics card driver is normal and is the latest version

![image](https://github.com/Alex313031/Thorium-Win/assets/72642350/a87638c5-7cbd-4792-b9e4-fa8aa773dc46)

**Comment by thesunnymachine** [comment]
Looks like it's only happening in Thorium when I tested today.  When I posted I tested with Edge and Chrome and it was causing issues in Thorium with YT playback, but today I can only get it to happen when Thorium has windows open for live streams. 

I recorded a video of the issue:  https://youtu.be/dEZ5PSg-vT4
It's still processing but you can see clearly that when I open the streams and close them that youtube lags / buffers and catches back up when I close the streams.

**Comment by Wolfsbane378** [comment]
I'm having the same issue with youtube in general on Thorium. Had to switch back to chrome

**Comment by Rairof** [comment]
YouTube playback on Fullscreen gives a lot of stutters with hardware acceleration on but disabling it solves the issue. Hope its fixed soon.

**Comment by gz83** [comment]
Please provide the link(s) to the problematic video(s) and I will try to reproduce the problem you are experiencing.

**Comment by Rairof** [comment]
https://github.com/Alex313031/Thorium-Win/issues/103#issuecomment-1794267954
It's all videos in full screen. You can watch any videos in full screen and it stutters with hardware acceleration on.

**Comment by gz83** [comment]
<img width="573" alt="Snipaste_2023-11-07_02-10-16" src="https://github.com/Alex313031/Thorium-Win/assets/17492865/d6b1fb00-ef72-4f33-a1a4-bd0e7313a601">

<img width="528" alt="Snipaste_2023-11-07_02-11-37" src="https://github.com/Alex313031/Thorium-Win/assets/17492865/be605991-481c-42fd-9bf9-4c3112552940">

I use Thorium on my computer to randomly play some videos in full-screen mode, using 1080P image quality. Under the default configuration, no abnormalities have been found (at least during the video playback process), and the progress bar can also be dragged at will.

**Comment by fubaWoW** [comment]
Same issue here, but...

This is not a Thorium only issue, also Edge have this issue but only if ``Hardware Acceleration`` is enabed on my System!
Very easy to test if you have a Unreal 5 Engine Game and "try" to run Youtube or Twitch on a 2nd Monitor for example while the Unreal 5 Game is running.
This will make ALL Browser Videos almost "stop" and stutter all the time.

Had to disable ``Hardware Acceleration`` and it works now.

My System btw:
CPU: Intel i7-13700k
RAM: Corsair DDR5 64GB 6600MHz CL32
GPU: RTX 3070Ti
...

**Comment by thesunnymachine** [comment]
> Same issue here, but...
> 
> This is not a Thorium only issue and it is definitely not just Intel! Also Edge have this issue but only if `Hardware Acceleration` is enabed on my System! Very easy to test if you have a Unreal 5 Engine Game and "try" to run Youtube or Twitch on a 2nd Monitor for example while the Unreal 5 Game is running. This will make ALL Browser Videos almost "stop" and stutter all the time.
> 
> Had to disable `Hardware Acceleration` and it works now.
> 
> My System btw: CPU: Intel i7-13700k RAM: Corsair DDR5 64GB 6600MHz CL32 GPU: Nvidia RTX 3070Ti ...

Interesting.  I am not having this issue with Chrome at the moment. 

Today, I tested this issue again and I am not seeing buffering, but I am seeing a lot more dropped frames and completely different stats in the "stats for nerds" info. 

Using the same exact video, here's my info from chrome:  
![image](https://github.com/Alex313031/Thorium-Win/assets/72642350/d3436170-014c-4131-87dd-7145fe5f08c0)

I only got dropped frames while I alt tabbed and setup to take the screen grab on chrome. 

With Thorium: 
![image](https://github.com/Alex313031/Thorium-Win/assets/72642350/113ca139-1e41-4ca6-9090-cec359bd19e9)
Here I was dropping frames pretty often, while just watching the video. 

On a third test, I got no dropped frames at all.  2 Streams open in other windows, no buffering no stuttering.  Maybe the Nvidia driver 546.01 update fixed the issue?  I will use thorium for a while again and see if the issue comes back.

**Comment by fubaWoW** [comment]
> Interesting. I am not having this issue with Chrome at the moment.

Yeah on Chrome i also so not have this issue, but on Edge, Thorium and Brave.
don't ask me why -> i don't know ^^

> On a third test, I got no dropped frames at all. 2 Streams open in other windows, no buffering no stuttering. Maybe the Nvidia driver 546.01 update fixed the issue? I will use thorium for a while again and see if the issue comes back.

Ok..
I have 546.08 installed
I will Test it later with Remnant 2, that Game forced all the Browsers i tested (except Chrome) to stutter on my 2nd monitor if ``Hardware Acceleration`` is enabled.
But i will also install and Test on Chrome again later!

**Comment by ElPrudi** [comment]
All videos and gifs flickered black constantly on Thorium. Disabling hardware acceleration definitely helped me.

**Comment by Consumaut** [comment]
Same here. Used to use Edge, Chrome and Firefox without any problems. What others describe only happened with Thorium.

Windows 11 23H2 (Build 22631.2715)
AMD Ryzen 9 5950x @4100 MHz
NVIDIA GeForce RTX 3090  546.29

I`m glad I found this threat, cause with Hardware Acceleration off everything works like a charm.
It feels like while playing the buffer it is not loading anything from the network till the buffer end is reached.
Happenms on Youtube and Twitch when plaiyng both in seperate windows on a widescreen at the same time, also when a video is in background while opening a new tab on youtube to search for another video.

**Comment by PrincessAkira** [comment]
Heya,
i have this aswell on AMD hardware. 7900XT on Windows.
latest thorium avx2 build.

**Comment by VerocityzWorld** [comment]
Same problem here, help :(

**Comment by cronner** [comment]
same problem fullscreen or no fullscreen, Chrome is fine

**Comment by ayuwooh** [comment]
Same or similar issue here. If Hardware Acceleration is on, videos on YT, Twitch and Threads will run like shit.

**Comment by Nagarafas** [comment]
I've faced this problem before. The solution is simple af.

- Open NVidia control panel
- Go to "Manage 3D settings"
- And turn off Multi-Frame Sampled AA (MFAA) in "Global Settings"

![Image](https://github.com/user-attachments/assets/23df9ee4-6fe5-4ef0-9ec8-a8a76a4955de)

Or go to "Program-Settings" and turn it off just for Thorium
- so just press add select Thorium and turn off MFAA there 

![Image](https://github.com/user-attachments/assets/376d58dd-0ffd-465b-918e-17567fbb2f09)

and do the same with "Background Application Max Frame Rate" (although i don't remember if that affects it or not)

Hope this helped :)

**Comment by ShufflePerson** [comment]
same issue, with hardware acceleration is enabled, any video playbacks are choppy.

**Comment by gz83** [comment]
Please try the latest version 138. If you still have problems, please let us know.
